### PR TITLE
Various version bumps and consequent fixes

### PR DESCRIPTION
**Pull request by milessabin**
+ Fixed stricter 2.13.5 linter warnings.
+ Updated travis build to 2.13.5.
+ Updated SBT to 1.4.9.

### CODE REVIEW

File: examples/src/main/scala/shapeless/examples/enum.scala [hunk 1]
```diff
@@ -37,9 +37,9 @@ object ScalaEnumDemo /*extends App*/ {
 
   // However ...
 
-  def isWeekend(d: WeekDay) = d match {
+  def isWeekend(d: WeekDay) = (d: @unchecked) match {
     case Sat | Sun => true
-    // Oops! Missing case ... still compiles
+    // Oops! Missing case ... still compiles (although as of 2.13.5 only with @unchecked)
```
Comment (joroKr21) [inline examples/src/main/scala/shapeless/examples/enum.scala:42]: Progress! 😄

**Review by joroKr21 — CHANGES_REQUESTED** [review]
Let's bump 2.12 as well.

We could disable `-Xlint:-valpattern` to avoid some of the exhaustivity warnings. WDYT?

File: examples/src/main/scala/shapeless/examples/deepsearch.scala [hunk 2]
```diff
@@ -82,10 +82,10 @@ object DeepSearchExamples extends App {
   assert(("a", ("b", "c"), "d").deepFind(p) == None)
 
   // On tuples with non-string elements:
-  assert((1, "two", (Symbol("three"), '4')).deepFind(p) == Some("two"))
+  assert((1, "two", (3.0, '4')).deepFind(p) == Some("two"))
```
[the remark this answers is not in the document]
Comment (milessabin) [reply examples/src/main/scala/shapeless/examples/deepsearch.scala:85]: I don't really either. It's some weirdness with `Symbol.apply` leading to match inexhaustivity, and given this is just a demo it didn't seem cost effective to dig any more.

File: .travis.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ jdk:
 scala:
   - 2.11.12
   - 2.12.12
```
[the remark this answers is not in the document]
Comment (milessabin) [reply .travis.yml:20]: :+1:

**Comment by milessabin** [comment]
I'd rather leave the exhaustiveness warnings in and disable them locally as I've done ... these are all oddities due to the way types and trees are defined in the compiler cake so it makes sense to special case them IMO.

**Review by joroKr21 — APPROVED** [review]
Nice 👍

**Pull request merged**